src/Microsoft.AspNet.Mvc.Core/Formatters/MediaTypeEncoding.cs

```diff
@@ -39,12 +38,21 @@ public static Encoding GetEncoding(string mediaType)
         /// <returns>The charset of the media type if it exists or a <see cref="StringSegment"/> without value if not.</returns>
         public static StringSegment GetCharsetParameter(StringSegment mediaType)
         {
-            MediaTypeHeaderValue parsedMediaType = null;
-            if (MediaTypeHeaderValue.TryParse(mediaType.Value, out parsedMediaType))
+            var parser = new MediaTypeParser(
+                mediaType.Buffer,
+                mediaType.Offset,
+                mediaType.Length);
+
+            var parameter = parser.GetParameter("charset");
+
+            if (parameter.HasValue)
+            {
+                return parameter.Value.Value;
+            }
+            else
             {
-                return new StringSegment(parsedMediaType.Charset);
+                return new StringSegment(buffer: null);
```

Comment:
Yes, I can change it if you want, but I feel like new StringSegment(buffer: null) represents the intent more clearly, I don't like to use default constructors in structs because it's indistinguishable that you initialized the type from it being initialized by default.